### PR TITLE
erlcloud_iam: fix get_account_authorization_details

### PR DESCRIPTION
**Pull request by nalundgaard**
When reviewing https://github.com/erlcloud/erlcloud/pull/671, I noticed that IAM [GetAccountAuthorizationDetails](https://docs.aws.amazon.com/IAM/latest/APIReference/API_GetAccountAuthorizationDetails.html) is a paginated API, and when you use it on an account with a paginated response, it crashes rather inelegantly.

This change makes it work correctly. `get_account_authorization_details_all` is left as an exercise for some other time.

### CODE REVIEW

File: src/erlcloud_iam.erl [hunk 1]
```diff
@@ -664,17 +664,25 @@ get_instance_profile(ProfileName, #aws_config{} = Config) ->
...
   
--spec get_account_authorization_details(aws_config()) -> {ok, proplist()} |  {error, any()}.
+-spec get_account_authorization_details(list() | aws_config()) -> {ok, proplist()} | {ok, proplist(), string()} | {error, any()}.
 get_account_authorization_details(#aws_config{} = Config) ->
+    get_account_authorization_details([], #aws_config{} = Config);
+get_account_authorization_details(Params) ->
+    get_account_authorization_details(Params, default_config()).
+
+-spec get_account_authorization_details(list(), aws_config()) -> {ok, proplist()} | {ok, proplist(), string()} | {error, any()}.
+get_account_authorization_details(Params, #aws_config{} = Config) ->
```
Comment (motobob) [inline src/erlcloud_iam.erl:678]: why not add a guard for a `when is_list(Params)` ?

**Review by motobob — APPROVED** [review]
suggesting a minor thing

**Review by kkuzmin — APPROVED** [review]
thank you @nalundgaard

File: src/erlcloud_iam.erl (hunk 1 above)
Comment (nalundgaard) [reply src/erlcloud_iam.erl]: Done.

**Pull request merged**